src/components/content/rich-text/rich-text.tsx

```diff
@@ -0,0 +1,330 @@
+        const { contentType, url } = node.data.target.file;
+        switch (contentType.split('/')[0]) {
+          /**
+           * Embedded images
+           */
+          case EmbeddedAssetType.IMAGE: {
+            return customComponents.figure({
+              children: [
+                <Zoom key={node.data.target.file.url}>
+                  <GatsbyImage
+                    image={node.data.target.gatsbyImageData}
+                    alt={node.data.target.description}
+                  />
+                </Zoom>,
+                customComponents.figcaption({
+                  key: 'figcaption',
+                  children: node.data.target.description,
+                }),
+              ],
+            });
+          }
+          /**
+           * Embedded videos
+           */
+          case EmbeddedAssetType.VIDEO: {
+            return customComponents.figure({
+              children: [
+                <video
+                  key="video"
+                  preload="auto"
+                  autoPlay
+                  loop
+                  muted
+                  width="100%"
+                >
+                  <source src={url} type={contentType} />
+                </video>,
+                customComponents.figcaption({
+                  key: 'figcaption',
+                  children: node.data.target.description,
+                }),
+              ],
+            });
+          }
+          default: {
+            console.error(`Content type is not implemented: ${contentType}`);
+            return null;
+          }
+        }
+      },
+    },
+  };
+
+  /**
+   * Footnote elements
+   */
+  const footNoteElements = customComponents.ol({
+    children: footnotes.map((footnote) => {
+      const { contentful_id, note } = footnote;
+      const { anchor, referenceAnchor } = footnoteReferences[contentful_id];
+      return (
+        <li key={contentful_id} id={anchor}>
+          <div style={{ display: 'flex', gap: 10 }}>
+            <MarkdownAst
+              key={contentful_id}
+              htmlAst={note.childMarkdownRemark.htmlAst}
+            />
+            {customComponents.a({
+              children: '\u21A9',
+              href: `#${referenceAnchor}`,
+              onClick: (e: MouseEvent) => {
+                e.preventDefault();
+                const target = document.querySelector(
+                  `[id='${referenceAnchor}']`,
+                );
+                history.pushState({}, '', `#${referenceAnchor}`);
+                scrollToTarget(target);
+              },
+            })}
+          </div>
+        </li>
+      );
+    }),
+  });
+
+  /**
+   * Rendered Rich Text + Footnotes
+   */
+  return (
+    <div>
+      {renderRichText(data, contentfulRenderOptions)}
+      {footnotes.length > 0 && (
+        <>
+          <hr />
+          {footNoteElements}
+        </>
+      )}
+    </div>
+  );
+};
+
+export const RichText = ({ data }: RichTextProps) => {
```

Comment:
Is this wrapper even necessary? Just a thought, but maybe the `customContentfulRenderer` can be a react component in the first place.